notebooks/Autoencoders.ipynb

```diff
@@ -0,0 +1,632 @@
+    "        # Fit the auto-encoder to predict X from X\n",
+    "        super().fit(X=X, y=X, sample_weight=sample_weight)\n",
+    "        # Copy encoder/decoder models to avoid modifying user input objects\n",
+    "        self.tf_est_ = copy.copy(self.tf_est)\n",
+    "        self.inv_tf_est_ = copy.copy(self.inv_tf_est)\n",
+    "        # Set params for encoder/decoder models\n",
+    "        self.tf_est_.set_params(fit_model=self.model_)\n",
+    "        self.inv_tf_est_.set_params(fit_model=self.model_, encoding_dim=self.encoding_dim)\n",
+    "        # Preprocess X\n",
+    "        X = self.feature_encoder_.transform(X)\n",
+    "        # Fit tf_est_\n",
+    "        self.tf_est_.initialize(X=X)\n",
+    "        X_tf = self.tf_est_.predict(X=X)\n",
+    "        # Fit inverse TF\n",
+    "        self.inv_tf_est_.initialize(X_tf)\n",
+    "        return self\n",
+    "\n",
+    "    def transform(self, X):\n",
+    "        X = self.feature_encoder_.transform(X)\n",
+    "        X_tf = self.tf_est_.predict(X)\n",
+    "        return X_tf\n",
+    "    \n",
+    "    def inverse_transform(self, X_tf):\n",
+    "        X = self.inv_tf_est_.predict(X_tf)\n",
+    "        X = self.feature_encoder_.inverse_transform(X)\n",
+    "        return X"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we wrap the Keras Model with Scikeras. Note that for our encoder/decoder estimators, we do not need to provide a loss function since no training will be done. We do however need to have the `fit_model` and `encoding_dim` so that these will be settable by `BaseWrapper.set_params`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tf_est = BaseWrapper(model=get_tf_model, fit_model=None, verbose=0)\n",
+    "inv_tf_est = BaseWrapper(model=get_inverse_tf_model, fit_model=None, encoding_dim=None, verbose=0)\n",
+    "autoencoder = KerasTransformer(model=get_fit_model, tf_est=tf_est, inv_tf_est=inv_tf_est, loss=\"binary_crossentropy\", encoding_dim=32, epochs=50)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "yqBl5xLDP43O"
+   },
+   "source": [
+    "## Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "4fHbt_mUVBhE"
+   },
+   "source": [
+    "To train the model, we pass the input images as both the features and the target. This will train the layers to compress the data as accurately as possible between the encoder and decoder. Note that we only pass the `X` parameter, since we defined the mapping `y=X` in `KerasTransformer.fit` above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 34
+    },
+    "id": "QQhtIWsMP4Ii",
+    "outputId": "1581384e-7012-4816-d4cf-a4190a30e50f"
+   },
+   "outputs": [
```

Comment:
Style nit: this is a lot of output, especially with the unnecessary printing of `Out[27]`. I think `autoencoder.fit(X=x_train);` or `_ = autoencoder.fit(X=x_train)` would remove the long output.

Reply:
Went with `_ = autoencoder.fit(X=x_train)`